llmfoundry/command_utils/train.py

```diff
@@ -640,7 +640,37 @@ def train(cfg: DictConfig) -> Trainer:
         trainer.eval()
 
     log.info('Starting training...')
-    trainer.fit()
+    try:
+        trainer.fit()
+    except ValueError as e:
+        msg = str(e)
+        if 'The max_duration' in msg and 'is less than or equal to the elapsed training duration' in msg and train_cfg.run_is_retry:
+            log.info(
+                'Training is already complete and detected retry. Skipping training and saving checkpoint.',
+            )
+            trainer.save_checkpoint_to_save_folder()
```

Comment:
i believe you don't want to call the composer save, just the hf save. because if the you get this error that means the last composer checkpoint was already saved successfully.